modules/launcher/src/main/java/org/mule/runtime/module/launcher/ApplicationClassLoaderBuilder.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.launcher;
+
+import static java.lang.String.format;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.descriptor.ArtifactDescriptor;
+import org.mule.runtime.module.launcher.domain.Domain;
+import org.mule.runtime.module.launcher.domain.DomainRepository;
+import org.mule.runtime.module.launcher.plugin.ArtifactPluginDescriptor;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * {@link ArtifactClassLoader} builder for class loaders required by {@link org.mule.runtime.module.launcher.application.Application} artifacts
+ *
+ * @since 4.0
+ */
+public class ApplicationClassLoaderBuilder
+{
+
+    private final ArtifactClassLoaderBuilder artifactClassLoaderBuilder;
+    private final DomainRepository domainRepository;
+    private String domain;
```

Comment:
Not sure about this, but think that default value should be "default" instead of null